src/Aspire.Hosting.Azure.SignalR/AzureSignalRExtensions.cs

```diff
@@ -59,4 +63,61 @@ public static IResourceBuilder<AzureSignalRResource> AddAzureSignalR(this IDistr
                       .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
                       .WithManifestPublishingCallback(resource.WriteToManifest);
     }
+
+    /// <summary>
+    /// Configures an Azure SignalR resource to be emulated. This resource requires an <see cref="AzureSignalRResource"/> to be added to the application model.
+    /// </summary>
+    /// <remarks>
+    /// This version of the package defaults to the <inheritdoc cref="SignalREmulatorContainerImageTags.Tag"/> tag of the <inheritdoc cref="SignalREmulatorContainerImageTags.Registry"/>/<inheritdoc cref="SignalREmulatorContainerImageTags.Image"/> container image.
+    /// </remarks>
+    /// <param name="builder">The Azure storage resource builder.</param>
+    /// <param name="configureContainer">Callback that exposes underlying container used for emulation to allow for customization.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureSignalRResource> RunAsEmulator(this IResourceBuilder<AzureSignalRResource> builder, Action<IResourceBuilder<AzureSignalREmulatorResource>>? configureContainer = null)
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return builder;
+        }
+
```

Comment:
shall we do some check here to make sure the SignalR service is in Serverless mode?
```csharp


        builder.ConfigureInfrastructure(i =>
        {
            // Get the SignalRService set by AddAzureSignalR
            var resource = i.GetProvisionableResources().OfType<SignalRService>().First(s => s.BicepIdentifier == i.AspireResource.GetBicepIdentifier());
            // check if the serviceMode is Serverless
            if (resource.Features.Any(feature => feature.Value != null && feature.Value.Flag.Value == SignalRFeatureFlag.ServiceMode && feature.Value.Value.Value != null && feature.Value.Value.Value.Equals("Serverless", StringComparison.OrdinalIgnoreCase)))
            {
                return;
            }
            else
            {
                throw new InvalidOperationException("Emulator works only for Azure SignalR Serverless mode");
            }
        });

```

Reply:
Seems like the current implementation of `AddAzureSignalR` would create a `Default` resource by default. So adding this change might break it